code/modules/food_and_drinks/food/snacks_pie.dm

```diff
@@ -28,33 +28,51 @@
 	tastes = list("pie" = 1)
 	foodtype = GRAIN | DAIRY | SUGAR
 	var/stunning = TRUE
+	var/experience_given = 10
+	var/experience_mod = 1
 
 /obj/item/reagent_containers/food/snacks/pie/cream/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
 	. = ..()
 	if(!.) //if we're not being caught
-		splat(hit_atom)
+		if(ishuman(throwingdatum.thrower))
+			var/mob/living/carbon/human/H
+			splat(hit_atom,H)
+		else
+			splat(hit_atom)
 
-/obj/item/reagent_containers/food/snacks/pie/cream/proc/splat(atom/movable/hit_atom)
+/obj/item/reagent_containers/food/snacks/pie/cream/proc/splat(atom/movable/hit_atom, mob/living/carbon/human/H)
 	if(isliving(loc)) //someone caught us!
 		return
 	var/turf/T = get_turf(hit_atom)
 	new/obj/effect/decal/cleanable/food/pie_smudge(T)
 	if(reagents && reagents.total_volume)
 		reagents.reaction(hit_atom, TOUCH)
-	if(isliving(hit_atom))
-		var/mob/living/L = hit_atom
-		if(stunning)
-			L.Paralyze(20) //splat!
-		L.adjust_blurriness(1)
-		L.visible_message("<span class='warning'>[L] is creamed by [src]!</span>", "<span class='userdanger'>You've been creamed by [src]!</span>")
-		playsound(L, "desceration", 50, TRUE)
 	if(is_type_in_typecache(hit_atom, GLOB.creamable))
 		hit_atom.AddComponent(/datum/component/creamed, src)
+	if(!isliving(hit_atom))
+		qdel(src)
+		return
+	var/mob/living/L = hit_atom
+	if(L.mind)
+		if(HAS_TRAIT(L.mind, TRAIT_LAW_ENFORCEMENT_METABOLISM))
+			experience_mod = 3
+		else
+			experience_mod = 1
+		H.mind.adjust_experience(/datum/skill/pie_throwing, experience_given * experience_mod)
+	if(stunning)
```

Comment:
Maybe allow clowns to stun people for a really short duration (0.5 seconds?) with syndicate pies if they're a legendary clown?

Reply:
syndi pies have same timers (only affected by skill)

Reply:
ah, I was thinking of the borg pies, sorry

speaking of which, borgs can't gain experience, right?